File: CalculatorApp/jmvvm-app/src/main/java/com/woowrale/calculator/japp/mvvm/CalculatorViewModel.java
package com.woowrale.calculator.japp.mvvm;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.woowrale.calculator.japp.usecase.CalculatorDivisionUseCase;
import com.woowrale.calculator.japp.usecase.CalculatorMinusUseCase;
import com.woowrale.calculator.japp.usecase.CalculatorMultiplyUseCase;
import com.woowrale.calculator.japp.usecase.CalculatorPlusUseCase;


public class CalculatorViewModel extends ViewModel {

    private MutableLiveData<Double> result;

    public CalculatorViewModel(){
        result = new MutableLiveData<>();
    }

    public LiveData<Double> getResult() {
        return result;
    }

    private void doPlusOperation(Double numberOne, Double numberTwo){
        result.setValue(new CalculatorPlusUseCase().executePlus(numberOne, numberTwo));
    }

    private void doMinusOperation(Double numberOne, Double numberTwo){
        result.setValue(new CalculatorMinusUseCase().executeMinus(numberOne, numberTwo));
    }

    private void doMultiplyOperation(Double numberOne, Double numberTwo){
        result.setValue(new CalculatorMultiplyUseCase().executeMultiply(numberOne, numberTwo));
    }

    private void doDivisionOperation(Double numberOne, Double numberTwo){
        result.setValue(new CalculatorDivisionUseCase().executeDivision(numberOne, numberTwo));
    }

    public void executeOperation(Double numberOne, Double numberTwo, String operator){
        switch (operator){
            case "+":
                doPlusOperation(numberOne, numberTwo);
                break;
            case "-":
                doMinusOperation(numberOne, numberTwo);
                break;
            case "*":
                doMultiplyOperation(numberOne, numberTwo);
                break;
            case "/":
                doDivisionOperation(numberOne, numberTwo);
                break;
        }
    }
}

File: CalculatorApp/kmvvm-app/src/main/java/com/woowrale/calculator/kapp/mvvm/CalculatorViewModel.kt
package com.woowrale.calculator.kapp.mvvm

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.woowrale.calculator.kapp.usecase.CalculatorDivisionUseCase
import com.woowrale.calculator.kapp.usecase.CalculatorMinusUseCase
import com.woowrale.calculator.kapp.usecase.CalculatorMultiplyUseCase
import com.woowrale.calculator.kapp.usecase.CalculatorPlusUseCase

class CalculatorViewModel : ViewModel() {
    private var result: MutableLiveData<Double>? = null

    init {
        result = MutableLiveData()
    }

    fun getResult(): LiveData<Double>? {
        return result
    }

    private fun doPlusOperation(
        numberOne: Double,
        numberTwo: Double
    ) {
        result?.value = CalculatorPlusUseCase().executePlus(numberOne, numberTwo)
    }

    private fun doMinusOperation(
        numberOne: Double,
        numberTwo: Double
    ) {
        result?.value = CalculatorMinusUseCase().executeMinus(numberOne, numberTwo)
    }

    private fun doMultiplyOperation(
        numberOne: Double,
        numberTwo: Double
    ) {
        result?.value = CalculatorMultiplyUseCase().executeMultiply(numberOne, numberTwo)
    }

    private fun doDivisionOperation(
        numberOne: Double,
        numberTwo: Double
    ) {
        result?.value = CalculatorDivisionUseCase().executeDivision(numberOne, numberTwo)
    }

    fun executeOperation(
        numberOne: Double,
        numberTwo: Double,
        operator: String?
    ) {
        when (operator) {
            "+" -> doPlusOperation(numberOne, numberTwo)
            "-" -> doMinusOperation(numberOne, numberTwo)
            "*" -> doMultiplyOperation(numberOne, numberTwo)
            "/" -> doDivisionOperation(numberOne, numberTwo)
        }
    }
}